### PR TITLE
PyPIConGPU compile test

### PR DESCRIPTION
**Pull request by SimeonEhrig**
Enable PyPIConGPU compile tests in the CI.

We test the compiling tests only with one configuration. This should be enough, because param files should be independent of the enabled backends.

Before this PR can be merged, the following PR can be merged:
- [x] #4575 
- [x] #4601

### CODE REVIEW

**Comment by BrianMarre** [comment]
this PR requires #4601 to be merged first

File: share/ci/install/pypicongpu.sh [hunk 1]
```diff
@@ -38,3 +50,34 @@ echo ""
 pip3 install -r $MODIFIED_REQUIREMENT_TXT
 cd $CI_PROJECT_DIR/test/python/picongpu
 python3 -m quick
+
+# executing the compiling tests is optinal
```
Comment (BrianMarre) [inline share/ci/install/pypicongpu.sh]: optinal --> optional

**Review by BrianMarre — APPROVED** [review]
one typo otherwise, good

File: share/ci/install/pypicongpu.sh (hunk 1 above)
Comment (SimeonEhrig) [reply share/ci/install/pypicongpu.sh]: fixed

**Comment by BrianMarre** [comment]
@psychocoderHPC ready for review

**Review by psychocoderHPC — APPROVED** [review]

**Pull request merged**